src/components/parcels/ParcelForm.svelte

```diff
@@ -0,0 +1,313 @@
+    parcelChannelDictionaryId = event.detail.id;
+  }
+
+  function onToggleCommandDictionary(event: CustomEvent) {
+    parcelCommandDictionaryId = event.detail.id;
+  }
+
+  function onToggleParameterDictionary(event: CustomEvent) {
+    selectedParmeterDictionaries = event.detail.ids;
+  }
+
+  function onToggleSequenceAdaptation(event: CustomEvent) {
+    parcelSequenceAdaptationId = event.detail.id;
+  }
+
+  async function saveParcelToParameterDictionaries(): Promise<void> {
+    const parcelToParameterDictionariesToAdd: Omit<ParcelToParameterDictionary, 'id'>[] = [];
+    const parcelToParameterDictionaryIdsToDelete: number[] = [];
+
+    Object.keys(selectedParmeterDictionaries).forEach(parameterDictionaryIdString => {
+      const parameterDictionaryId = parseInt(parameterDictionaryIdString);
+      const isSelected = selectedParmeterDictionaries[parameterDictionaryId];
+
+      if (!isSelected && savedParameterDictionaryIds[parameterDictionaryId]) {
+        // Parameter dictionary was removed from the parcel.
+        parcelToParameterDictionaryIdsToDelete.push(parameterDictionaryId);
+      } else if (isSelected && !savedParameterDictionaryIds[parameterDictionaryId] && parcelId) {
+        // Parameter dictionary was freshly added to the parcel and hasn't been saved yet.
+        parcelToParameterDictionariesToAdd.push({
+          parameter_dictionary_id: parameterDictionaryId,
+          parcel_id: parcelId,
+        });
+      }
+    });
+
+    if (parcelToParameterDictionariesToAdd.length > 0) {
+      await effects.createParcelToParameterDictionaries(parcelOwner, parcelToParameterDictionariesToAdd, user);
+    }
+
+    if (parcelToParameterDictionaryIdsToDelete.length > 0) {
+      const idsToDelete = [];
+
+      for (const paramDictionaryId of parcelToParameterDictionaryIdsToDelete) {
+        const parcelId: number | undefined = $parcelToParameterDictionaries.find(
+          p => p.parameter_dictionary_id === paramDictionaryId && p.parcel_id === initialParcelId,
+        )?.id;
+
+        if (parcelId) {
+          idsToDelete.push(parcelId);
+        }
+
+        if (idsToDelete.length > 0 && $parcel) {
+          await effects.deleteParcelToParameterDictionaries(idsToDelete, $parcel, user);
+        }
+      }
+    }
+  }
+
+  async function saveParcel() {
+    if (saveButtonEnabled) {
+      savingParcel = true;
+
+      if (parcelCommandDictionaryId !== null && parcelName !== '') {
+        if (mode === 'create') {
+          const newParcel: ParcelInsertInput = {
+            channel_dictionary_id: parcelChannelDictionaryId,
+            command_dictionary_id: parcelCommandDictionaryId,
+            name: parcelName,
+            sequence_adaptation_id: parcelSequenceAdaptationId,
+          };
+          parcelId = await effects.createParcel(newParcel, user);
+
+          await saveParcelToParameterDictionaries();
+
+          if (parcelId !== null) {
+            goto(`${base}/parcels/edit/${parcelId}`);
```

Comment:
Can we move this out of this component into the parent? This component can dispatch some saved event, that the parent can be listening for.